backend.go

```diff
@@ -21,6 +21,22 @@ type Backend interface {
 	AnonymousLogin(state *ConnectionState) (Session, error)
 }
 
+type Feature int32
+
+const (
+	FeatureSMTPUTF8 Feature = 2 << iota
+	FeatureBINARYMIME
+	FeatureREQUIRETLS
+)
+
+func (f Feature) Contains(feat Feature) bool {
+	return f&feat != 0
+}
+
+type FeatureBackend interface {
```

Comment:
Maybe embed the `Backend` interface here, to make it clear it's a backend extension

Reply:
Done.